Wikipedia/Code/WMFAppViewController.m

```diff
@@ -716,6 +718,12 @@ - (void)runDataMigrationIfNeededWithCompletion:(dispatch_block_t)completion {
 #pragma mark - UITabBarControllerDelegate
 
 - (void)tabBarController:(UITabBarController*)tabBarController didSelectViewController:(UIViewController*)viewController {
+    NSInteger prevActiveTabIndex = [[NSUserDefaults standardUserDefaults] integerForKey:PrevActiveTabIndex];
+    [[NSUserDefaults standardUserDefaults] setInteger:tabBarController.selectedIndex forKey:PrevActiveTabIndex];
+    if (tabBarController.selectedIndex == WMFAppTabTypeExplore && prevActiveTabIndex==WMFAppTabTypeExplore) {
+        WMFExploreViewController *exploreViewController = (WMFExploreViewController *)[self exploreViewController];
+        [exploreViewController.tableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0] atScrollPosition:UITableViewScrollPositionTop animated:YES];
```

Comment:
Generally we try not to touch the internals of another class and favor providing a specific method to accomplish a task - in this case it would be more encapsulated if we provided a public method like:

```
- (void)scrollToTop:(BOOL)animated;
```

And then we can let the explore view controller do the work as it seed fit.